### PR TITLE
M2-441: drawing-crop

### PR DESCRIPTION
**Pull request by kdanilenka32**

### CODE REVIEW

**Review by alex-i-e — APPROVED** [review]

**Review by nmialik — APPROVED** [review]

File: src/shared/components/Uploader/Uploader.tsx [hunk 1]
```diff
@@ -61,8 +71,32 @@ export const Uploader = ({
...
+  const checkDimensions = (file: File): Promise<boolean> =>
+    new Promise((resolve, reject) => {
+      const reader = new FileReader();
+
+      reader.onload = (e: ProgressEvent<FileReader>) => {
+        const img = new Image();
+        img.onload = () => {
+          const { width, height } = img;
+          resolve(
+            width > MIN_IMAGE_WIDTH &&
```
Comment (IhorNesterenko99) [inline src/shared/components/Uploader/Uploader.tsx:83]: maybe add equal also?
for example width >= MIN_IMAGE_WIDTH && ...

File: src/shared/components/Uploader/Uploader.tsx [hunk 2]
```diff
@@ -61,8 +71,32 @@ export const Uploader = ({
...
+              width < MAX_IMAGE_WIDTH &&
+              height > MIN_IMAGE_HEIGHT &&
+              height < MAX_IMAGE_HEIGHT,
+          );
+        };
+        img.onerror = reject;
+        img.src = e.target?.result as string;
```
Comment (IhorNesterenko99) [inline src/shared/components/Uploader/Uploader.tsx]:
```suggestion
        img.src = (e.target?.result || '') as string;
```

File: src/shared/components/Uploader/Uploader.tsx [hunk 3]
```diff
@@ -61,8 +71,32 @@ export const Uploader = ({
     e.preventDefault();
   };
 
+  const checkDimensions = (file: File): Promise<boolean> =>
+    new Promise((resolve, reject) => {
+      const reader = new FileReader();
+
+      reader.onload = (e: ProgressEvent<FileReader>) => {
```
Comment (IhorNesterenko99) [inline src/shared/components/Uploader/Uploader.tsx:78]:
```suggestion
      reader.onload = (event: ProgressEvent<FileReader>) => {
```

File: src/shared/components/Uploader/Uploader.tsx (hunk 1 above)
Comment (kdanilenka32) [reply src/shared/components/Uploader/Uploader.tsx:83]: By req it is between

File: src/shared/components/Uploader/Uploader.tsx (hunk 3 above)
Comment (kdanilenka32) [reply src/shared/components/Uploader/Uploader.tsx:78]: e it is ok))

**Pull request merged**